### PR TITLE
Remove async/await and some mocks from tests

### PR DESCRIPTION
**Pull request by hmhealey**
async/await and this mocking seemed to cause issues after this was moved into the web app. I'm not 100% sure that these will solve those issues, but they seem decent enough to clean up anyway.

### CODE REVIEW

**Comment by lieut-data** [comment]
To confirm, are all these `async` / `await` spurious, in that none of the methods return promises?

File: test/test_helper.js [hunk 1]
```diff
@@ -453,7 +453,7 @@ class TestHelper {
         this.basicGroup = this.fakeGroupWithId();
     }
 
-    initBasic = async (client4 = this.createClient4()) => {
+    initBasic = (client4 = this.createClient4()) => {
```
Comment (lieut-data) [inline test/test_helper.js:456]: Oh, I see we've actually stripped this here -- that makes more sense now!

**Review by lieut-data — APPROVED** [review]
Reviewed more carefully and realized we're stripping it holistically -- nice!

File: test/test_helper.js (hunk 1 above)
Comment (hmhealey) [reply test/test_helper.js:456]: Yeah, I don't know why it was even there to begin with. It was probably something we needed at some point, but never ended up using after that. We have had a tendency to overuse async/await frequently in this code.

**Review by nevyangelova — APPROVED** [review]
LGTM not sure why some of those were there to begin with though? Are we sure its not needed somewhere, maybe a promise is returned somewhere?

**Comment by hmhealey** [comment]
I think we could've used a promise somewhere, but it doesn't seem necessary any more at least. I'd prefer to simplify things by removing the async/awaits wherever is possible since it makes integrating the tests with the webapp code easier.

**Pull request merged**